src/main/java/com/minecolonies/colony/buildings/BuildingGuardTower.java

```diff
@@ -167,11 +370,152 @@ public int getBonusVision()
         return MAX_VISION_BONUS_MULTIPLIER * VISION_BONUS;
     }
 
+    @Override
+    public void readFromNBT(@NotNull final NBTTagCompound compound)
+    {
+        super.readFromNBT(compound);
+        task = Task.values()[compound.getInteger(TAG_TASK)];
+        int jobId = compound.getInteger(TAG_JOB);
+        job = jobId == -1 ? null : GuardJob.values()[jobId];
+        assignManually = compound.getBoolean(TAG_ASSIGN);
+        retrieveOnLowHealth = compound.getBoolean(TAG_RETRIEVE);
+        patrolManually = compound.getBoolean(TAG_PATROL);
+
+        final NBTTagList wayPointTagList = compound.getTagList(TAG_PATROL_TARGETS, Constants.NBT.TAG_COMPOUND);
+        for (int i = 0; i < wayPointTagList.tagCount(); ++i)
+        {
+            NBTTagCompound blockAtPos = wayPointTagList.getCompoundTagAt(i);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'blockAtPos' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)